src/components/CurrencyInputPanel/CurrencyInputPanel.container.tsx

```diff
@@ -0,0 +1,97 @@
+import debounce from 'lodash.debounce'
+
+import { CurrencyInputPanelComponent } from './CurrencyInputPanel.component'
+
+import { CurrencySearchContext } from '../SearchModal/CurrencySearch/CurrencySearch.context'
+import { ListRowContext, ManageListsContext } from '../SearchModal/ManageLists/ManageLists.context'
+import { CurrencySearchModalContext } from '../SearchModal/CurrencySearchModal/CurrencySearchModal.context'
+
+import {
+  useCurrencySearchModalBridge,
+  useCurrencySearchModalSwap
+} from '../SearchModal/CurrencySearchModal/CurrencySearchModal.hooks'
+import { useBridgeInputValidation } from '../../pages/Bridge/ActionPanel/useBridgeInputValidation'
+
+import { CurrencyWrapperSource } from '../CurrencyLogo'
+import { CurrencyInputPanelProps } from './CurrencyInputPanel.types'
+
+export const CurrencyInputPanel = (currencyInputPanelProps: CurrencyInputPanelProps) => {
+  const {
+    manageListsContext,
+    listRowContext,
+    currencySearchContext,
+    currencySearchModalContext
+  } = useCurrencySearchModalSwap()
+
+  return (
+    <CurrencySearchModalContext.Provider value={currencySearchModalContext}>
+      <CurrencySearchContext.Provider value={currencySearchContext}>
+        <ManageListsContext.Provider value={manageListsContext}>
+          <ListRowContext.Provider value={listRowContext}>
+            <CurrencyInputPanelComponent {...currencyInputPanelProps} />
+          </ListRowContext.Provider>
+        </ManageListsContext.Provider>
+      </CurrencySearchContext.Provider>
+    </CurrencySearchModalContext.Provider>
+  )
+}
+
+export const CurrencyInputPanelBridge = (currencyInputPanelProps: CurrencyInputPanelProps) => {
+  const {
+    manageListsContext,
+    listRowContext,
+    currencySearchContext,
+    currencySearchModalContext
+  } = useCurrencySearchModalBridge()
+
+  const {
+    value: valueRaw,
+    onUserInput: onUserInputRaw,
+    displayedValue,
+    setDisplayedValue,
+    disableCurrencySelect
+  } = currencyInputPanelProps
+
+  const debounceOnUserInput = useMemo(() => {
+    return debounce(onUserInputRaw, 500)
+  }, [onUserInputRaw])
+
+  const onUserInput = useCallback(
+    (val: string) => {
+      setDisplayedValue && setDisplayedValue(val)
+      debounceOnUserInput(val)
+    },
+    [debounceOnUserInput, setDisplayedValue]
+  )
+
+  const value = useMemo(() => (disableCurrencySelect ? valueRaw : displayedValue ?? ''), [
+    disableCurrencySelect,
+    displayedValue,
+    valueRaw
+  ])
+
+  useEffect(() => {
+    debounceOnUserInput.cancel()
+  }, [debounceOnUserInput, disableCurrencySelect])
+
+  useBridgeInputValidation(!!disableCurrencySelect)
+
+  return (
+    <CurrencySearchModalContext.Provider value={currencySearchModalContext}>
+      <CurrencySearchContext.Provider value={currencySearchContext}>
+        <ManageListsContext.Provider value={manageListsContext}>
+          <ListRowContext.Provider value={listRowContext}>
```

Comment:
I see this is repeating. Should we define a `CurrencyInputContext` and use it in `CurrencyInputPanel` and `CurrencyInputPanelBridge`?

Reply:
Yep, good catch. Please look at `CurrencySearchModalProvider`